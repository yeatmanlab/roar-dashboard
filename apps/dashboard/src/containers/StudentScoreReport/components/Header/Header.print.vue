<template>
  <div class="text-sm">
    <div class="text-left">
      <h1 class="p-0 m-0 text-5xl">
        <span class="block mb-1 text-sm text-gray-600 uppercase">
          {{ $t('scoreReports.pageTitle') }}
        </span>
        <span class="hidden">: </span>
        <span>{{ studentFirstName }} {{ studentLastName }}</span>
      </h1>
    </div>

    <dl v-if="studentGrade || className || administrationName" class="flex gap-6 justify-between py-2 w-full">
      <div class="flex gap-6">
        <div v-if="administrationName" class="inline-block">
          <dt class="inline font-semibold">{{ $t('scoreReports.administration') }}:</dt>
          <dd class="inline ml-2">{{ administrationName }}</dd>
        </div>
        <div v-if="className" class="inline-block">
          <dt class="inline font-semibold">{{ $t('scoreReports.class') }}:</dt>
          <dd class="inline ml-2">{{ className }}</dd>
        </div>
      </div>

      <div v-if="studentGrade" class="inline-block">
        <dt class="inline font-semibold">{{ $t('scoreReports.grade') }}:</dt>
        <dd class="inline ml-2">{{ getGradeWithSuffix(studentGrade) }}</dd>
      </div>
    </dl>
  </div>
</template>

<script setup>
import { getGradeWithSuffix } from '@/helpers/reports.js';

defineProps({
  studentFirstName: {
    type: String,
    required: true,
  },
  studentLastName: {
    type: String,
    default: '',
  },
  studentGrade: {
    type: [String, Number],
    default: null,
  },
  className: {
    type: String,
    default: '',
  },
  administrationName: {
    type: String,
    default: '',
  },
  expanded: {
    type: Boolean,
    default: false,
  },
  exportLoading: {
    type: Boolean,
    default: false,
  },
});

defineEmits(['toggleExpand', 'exportPdf']);
</script>
