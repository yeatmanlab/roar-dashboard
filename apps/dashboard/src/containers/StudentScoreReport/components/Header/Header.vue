<template>
  <section data-cy="report__header" aria-labelledby="page-title">
    <div class="flex flex-column lg:flex-row lg:align-items-center lg:justify-between">
      <h1 id="page-title" class="p-0 m-0 text-5xl">
        <span class="block mb-1 text-sm font-medium text-gray-500 uppercase">
          {{ $t('scoreReports.pageTitle') }}
        </span>
        <span>{{ studentFirstName }} {{ studentLastName }}</span>
      </h1>

      <div class="flex gap-2">
        <PvButton
          severity="secondary"
          variant="outlined"
          :label="!expanded ? $t('scoreReports.expandSections') : $t('scoreReports.collapseSections')"
          :icon="
            !expanded ? 'pi pi-chevron-down ml-2 transition-all' : 'pi pi-chevron-down ml-2 transition-all rotate-180'
          "
          icon-pos="right"
          data-html2canvas-ignore="true"
          data-cy="report__expand-btn"
          @click="$emit('toggleExpand')"
        />

        <PvButton
          severity="secondary"
          variant="outlined"
          :label="$t('scoreReports.exportPDF')"
          :icon="exportLoading ? 'pi pi-spin pi-spinner ml-2' : 'pi pi-download ml-2'"
          :disabled="exportLoading"
          icon-pos="right"
          data-html2canvas-ignore="true"
          data-cy="report__pdf-export-btn"
          @click="$emit('exportPdf')"
        />
      </div>
    </div>

    <dl v-if="studentGrade || className || administrationName" class="flex flex-wrap gap-4 mt-4 w-full text-gray-600">
      <div v-if="studentGrade" class="inline-flex p-1 px-2 bg-gray-200 rounded">
        <dt class="font-medium">{{ $t('scoreReports.grade') }}:</dt>
        <dd class="ml-1">{{ getGradeWithSuffix(studentGrade) }}</dd>
      </div>
      <div v-if="className" class="inline-flex p-1 px-2 bg-gray-200 rounded">
        <dt class="font-medium">{{ $t('scoreReports.class') }}:</dt>
        <dd class="ml-1">{{ className }}</dd>
      </div>
      <div v-if="administrationName" class="inline-flex p-1 px-2 bg-gray-200 rounded">
        <dt class="font-medium">{{ $t('scoreReports.administration') }}:</dt>
        <dd class="ml-1">{{ administrationName }}</dd>
      </div>
    </dl>
  </section>
</template>

<script setup>
import PvButton from 'primevue/button';
import { getGradeWithSuffix } from '@/helpers/reports.js';

defineProps({
  studentFirstName: {
    type: String,
    required: true,
  },
  studentLastName: {
    type: String,
    default: '',
  },
  studentGrade: {
    type: [String, Number],
    default: null,
  },
  className: {
    type: String,
    default: '',
  },
  administrationName: {
    type: String,
    default: '',
  },
  expanded: {
    type: Boolean,
    default: false,
  },
  exportPdf: {
    type: Function,
    default: () => {},
  },
  exportLoading: {
    type: Boolean,
    default: false,
  },
});

defineEmits(['toggleExpand', 'exportPdf']);
</script>
