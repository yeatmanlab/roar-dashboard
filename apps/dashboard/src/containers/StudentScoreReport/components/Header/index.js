export { default as HeaderScreen } from './Header.vue';
export { default as HeaderPrint } from './Header.print.vue';
