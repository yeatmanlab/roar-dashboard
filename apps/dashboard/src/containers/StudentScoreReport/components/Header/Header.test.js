import { describe, it, expect, vi, beforeEach } from 'vitest';
import { mount } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';
import Header from './Header.vue';

// Mock the getGradeWithSuffix helper
vi.mock('@/helpers/reports.js', () => ({
  getGradeWithSuffix: vi.fn((grade) => {
    const gradeNum = parseInt(grade);
    if (gradeNum === 0) return 'Kindergarten';
    const suffix = gradeNum === 1 ? 'st' : gradeNum === 2 ? 'nd' : gradeNum === 3 ? 'rd' : 'th';
    return `${gradeNum}${suffix} Grade`;
  }),
}));

describe('Header.vue - Snapshots', () => {
  let i18n;

  beforeEach(() => {
    i18n = createI18n({
      legacy: false,
      locale: 'en',
      messages: {
        en: {
          scoreReports: {
            pageTitle: 'Student Score Report',
            expandSections: 'Expand All',
            collapseSections: 'Collapse All',
            exportPDF: 'Export PDF',
            grade: 'Grade',
            class: 'Class',
            administration: 'Administration',
          },
        },
      },
    });
  });

  describe('snapshots', () => {
    it('renders minimal props (firstName only)', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders with full props', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          studentLastName: 'Doe',
          studentGrade: 3,
          className: 'Ms. Smith',
          administrationName: 'Fall 2024',
          expanded: false,
          exportLoading: false,
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders expanded state', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          studentLastName: 'Doe',
          expanded: true,
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders loading state', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          exportLoading: true,
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders with Kindergarten grade', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'Sarah',
          studentGrade: 0,
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without optional metadata', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          studentLastName: 'Doe',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('component behavior', () => {
    it('displays student name correctly', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          studentLastName: 'Doe',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.text()).toContain('John Doe');
    });

    it('emits toggleExpand event when expand button clicked', async () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
        },
        global: {
          plugins: [i18n],
        },
      });

      const expandBtn = wrapper.find('[data-cy="report__expand-btn"]');
      await expandBtn.trigger('click');

      expect(wrapper.emitted('toggleExpand')).toHaveLength(1);
    });

    it('emits exportPdf event when export button clicked', async () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
        },
        global: {
          plugins: [i18n],
        },
      });

      const exportBtn = wrapper.find('[data-cy="report__pdf-export-btn"]');
      await exportBtn.trigger('click');

      expect(wrapper.emitted('exportPdf')).toHaveLength(1);
    });

    it('disables export button when loading', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          exportLoading: true,
        },
        global: {
          plugins: [i18n],
        },
      });

      const exportBtn = wrapper.find('[data-cy="report__pdf-export-btn"]');
      expect(exportBtn.attributes('disabled')).toBeDefined();
    });

    it('shows loading spinner when exportLoading is true', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          exportLoading: true,
        },
        global: {
          plugins: [i18n],
        },
      });

      const exportBtn = wrapper.find('[data-cy="report__pdf-export-btn"]');
      expect(exportBtn.html()).toContain('pi-spinner');
    });

    it('renders with expanded prop', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          expanded: true,
        },
        global: {
          plugins: [i18n],
        },
      });

      // Just verify the component renders with the expanded prop
      expect(wrapper.exists()).toBe(true);
      expect(wrapper.props('expanded')).toBe(true);
    });

    it('displays grade metadata when provided', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          studentGrade: 5,
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.text()).toContain('5th Grade');
    });

    it('displays class metadata when provided', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          className: 'Ms. Johnson',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.text()).toContain('Ms. Johnson');
    });

    it('displays administration metadata when provided', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
          administrationName: 'Spring 2024',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.text()).toContain('Spring 2024');
    });

    it('does not render metadata section when no metadata provided', () => {
      const wrapper = mount(Header, {
        props: {
          studentFirstName: 'John',
        },
        global: {
          plugins: [i18n],
        },
      });

      // Check that the dl element (metadata container) doesn't exist
      expect(wrapper.find('dl').exists()).toBe(false);
    });
  });
});
