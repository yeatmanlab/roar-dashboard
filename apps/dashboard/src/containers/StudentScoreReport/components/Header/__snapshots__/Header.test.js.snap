// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header.vue - Snapshots > snapshots > renders expanded state 1`] = `
"<section data-cy="report__header" aria-labelledby="page-title">
  <div class="flex flex-column lg:flex-row lg:align-items-center lg:justify-between">
    <h1 id="page-title" class="p-0 m-0 text-5xl"><span class="block mb-1 text-sm font-medium text-gray-500 uppercase"></span><span>John Doe</span></h1>
    <div class="flex gap-2"><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__expand-btn" pc0="" data-pc-section="root"><span class="p-button-icon pi pi-chevron-down ml-2 transition-all rotate-180" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__pdf-export-btn" pc1="" data-pc-section="root"><span class="p-button-icon pi pi-download ml-2" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button></div>
  </div>
  <!--v-if-->
</section>"
`;

exports[`Header.vue - Snapshots > snapshots > renders loading state 1`] = `
"<section data-cy="report__header" aria-labelledby="page-title">
  <div class="flex flex-column lg:flex-row lg:align-items-center lg:justify-between">
    <h1 id="page-title" class="p-0 m-0 text-5xl"><span class="block mb-1 text-sm font-medium text-gray-500 uppercase"></span><span>John </span></h1>
    <div class="flex gap-2"><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__expand-btn" pc0="" data-pc-section="root"><span class="p-button-icon pi pi-chevron-down ml-2 transition-all" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" disabled="" data-pc-name="button" data-p-disabled="true" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__pdf-export-btn" pc1="" data-pc-section="root"><span class="p-button-icon pi pi-spin pi-spinner ml-2" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button></div>
  </div>
  <!--v-if-->
</section>"
`;

exports[`Header.vue - Snapshots > snapshots > renders minimal props (firstName only) 1`] = `
"<section data-cy="report__header" aria-labelledby="page-title">
  <div class="flex flex-column lg:flex-row lg:align-items-center lg:justify-between">
    <h1 id="page-title" class="p-0 m-0 text-5xl"><span class="block mb-1 text-sm font-medium text-gray-500 uppercase"></span><span>John </span></h1>
    <div class="flex gap-2"><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__expand-btn" pc0="" data-pc-section="root"><span class="p-button-icon pi pi-chevron-down ml-2 transition-all" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__pdf-export-btn" pc1="" data-pc-section="root"><span class="p-button-icon pi pi-download ml-2" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button></div>
  </div>
  <!--v-if-->
</section>"
`;

exports[`Header.vue - Snapshots > snapshots > renders with Kindergarten grade 1`] = `
"<section data-cy="report__header" aria-labelledby="page-title">
  <div class="flex flex-column lg:flex-row lg:align-items-center lg:justify-between">
    <h1 id="page-title" class="p-0 m-0 text-5xl"><span class="block mb-1 text-sm font-medium text-gray-500 uppercase"></span><span>Sarah </span></h1>
    <div class="flex gap-2"><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__expand-btn" pc0="" data-pc-section="root"><span class="p-button-icon pi pi-chevron-down ml-2 transition-all" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__pdf-export-btn" pc1="" data-pc-section="root"><span class="p-button-icon pi pi-download ml-2" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button></div>
  </div>
  <!--v-if-->
</section>"
`;

exports[`Header.vue - Snapshots > snapshots > renders with full props 1`] = `
"<section data-cy="report__header" aria-labelledby="page-title">
  <div class="flex flex-column lg:flex-row lg:align-items-center lg:justify-between">
    <h1 id="page-title" class="p-0 m-0 text-5xl"><span class="block mb-1 text-sm font-medium text-gray-500 uppercase"></span><span>John Doe</span></h1>
    <div class="flex gap-2"><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__expand-btn" pc0="" data-pc-section="root"><span class="p-button-icon pi pi-chevron-down ml-2 transition-all" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__pdf-export-btn" pc1="" data-pc-section="root"><span class="p-button-icon pi pi-download ml-2" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button></div>
  </div>
  <dl class="flex flex-wrap gap-4 mt-4 w-full text-gray-600">
    <div class="inline-flex p-1 px-2 bg-gray-200 rounded">
      <dt class="font-medium">:</dt>
      <dd class="ml-1">3rd Grade</dd>
    </div>
    <div class="inline-flex p-1 px-2 bg-gray-200 rounded">
      <dt class="font-medium">:</dt>
      <dd class="ml-1">Ms. Smith</dd>
    </div>
    <div class="inline-flex p-1 px-2 bg-gray-200 rounded">
      <dt class="font-medium">:</dt>
      <dd class="ml-1">Fall 2024</dd>
    </div>
  </dl>
</section>"
`;

exports[`Header.vue - Snapshots > snapshots > renders without optional metadata 1`] = `
"<section data-cy="report__header" aria-labelledby="page-title">
  <div class="flex flex-column lg:flex-row lg:align-items-center lg:justify-between">
    <h1 id="page-title" class="p-0 m-0 text-5xl"><span class="block mb-1 text-sm font-medium text-gray-500 uppercase"></span><span>John Doe</span></h1>
    <div class="flex gap-2"><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__expand-btn" pc0="" data-pc-section="root"><span class="p-button-icon pi pi-chevron-down ml-2 transition-all" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button><button class="p-button p-component p-button-icon-only p-button-secondary p-button-outlined" data-p="icon-only outlined" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" data-cy="report__pdf-export-btn" pc1="" data-pc-section="root"><span class="p-button-icon pi pi-download ml-2" data-p="right" data-pc-section="icon"></span>
        <!---->
        <!---->
      </button></div>
  </div>
  <!--v-if-->
</section>"
`;
